aws_google_auth/google.py

```diff
@@ -205,9 +207,27 @@ def do_login(self):
 
         self.check_extra_step(response_page)
 
+        # Process Google CAPTCHA verification request if present
         if cap is not None:
-            raise ExpectedGoogleException(
-                'Captcha Required. Manually Login to remove this.')
+            self.session.headers['Referer'] = sess.url
```

Comment:
I suppose more of this can be moved into `handle_captcha()`? If anybody wants to take a crack.

Reply:
I'm trying to understand the PR, it looks like it duplicates a bunch of code - and I can't tell if that's intentional or not.

Reply:
I think most of the code is from here: https://github.com/cevoaustralia/aws-google-auth/pull/91/commits/f00c65fdff1dbe9031b7782ca59129af86069753

And my PRs have moved @louahola code from `do_login()` into `handle_captcha()`

Reply:
Yep, I'm not commenting on your PR's as those have merged in here. I'm just trying to understand this PR in it's current state. I will take a look shortly.